fhir-search/src/main/java/com/ibm/fhir/search/util/SearchUtil.java

```diff
@@ -1589,6 +1605,168 @@ private static QueryParameter parseChainedParameter(Class<?> resourceType, Strin
+     * @param reverseChainParameterString
+     *          Reverse chain search parameter string.
+     * @param valuesString
+     *          String containing the final search value.
+     * @return QueryParameter
+     *          The root of a parameter chain for the reverse chain criteria.
+     */
+    private static QueryParameter parseReverseChainedParameter(Class<?> resourceType, String reverseChainParameterString, String valuesString) throws Exception {
+
+        QueryParameter rootParameter = null;
+
+        try {
+            // Strip leading '_has:' and then split by ':_has:'
+            List<String> components = Arrays.asList(reverseChainParameterString
+                .replaceFirst(SearchConstants.HAS + SearchConstants.COLON_DELIMITER_STR, "")
+                .split(SearchConstants.COLON_DELIMITER_STR + SearchConstants.HAS + SearchConstants.COLON_DELIMITER_STR));
+
+            if (components.size() == 0) {
+                throw SearchExceptionUtil.buildNewInvalidSearchException(INCORRECT_NUMBER_OF_COMPONENTS_FOR_REVERSE_CHAIN_SEARCH);
+            }
+
+            int currentIndex = 0;
+            int lastIndex = components.size() - 1;
+
+            for (String component : components) {
+                // Split into subcomponents by colon delimiter
+                List<String> subcomponents = Arrays.asList(component.split(SearchConstants.COLON_DELIMITER_STR, 3));
+
+                // Validate correct number of subcomponents
+                if ((currentIndex < lastIndex && subcomponents.size() != 2) ||
+                        (currentIndex == lastIndex && subcomponents.size() != 3)) {
+                    throw SearchExceptionUtil.buildNewInvalidSearchException(INCORRECT_NUMBER_OF_COMPONENTS_FOR_REVERSE_CHAIN_SEARCH);
+                }
+
+                // Validate referenced-by resource type
+                String referencedByResourceTypeName = subcomponents.get(0);
+                Class<? extends Resource> referencedByResourceType = ModelSupport.getResourceType(referencedByResourceTypeName);
+                if (referencedByResourceType == null) {
+                    throw SearchExceptionUtil.buildNewInvalidSearchException(
+                        String.format(INVALID_RESOURCE_TYPE_FOR_REVERSE_CHAIN_SEARCH, referencedByResourceTypeName));
+                }
+
+                // Validate reference search parameter
+                String referenceSearchParameterName = subcomponents.get(1);
+                SearchParameter referenceSearchParameter = getSearchParameter(referencedByResourceType, referenceSearchParameterName);
+                if (referenceSearchParameter == null) {
+                    throw SearchExceptionUtil.buildNewInvalidSearchException(
+                        String.format(SEARCH_PARAMETER_NOT_FOUND, referenceSearchParameterName, referencedByResourceTypeName));
+                }
+                if (!Type.REFERENCE.equals(Type.fromValue(referenceSearchParameter.getType().getValue()))) {
```

Comment:
at this point in the code would referenceSearchParameter.getType ever be null? 
I don't think so at this point in the model this will be null as the `type | Σ | 1..1 | code`

This is good.  Leaving my note here.